sokutei.h

```diff
@@ -0,0 +1,259 @@
+    int i = 0;
+    while(string_a[i] == string_b[i] && string_a[i]) i++;
+    return string_a[i] - string_b[i];
+}
+
+
+char *sokutei_strcpy(char *string_a, const char *string_b){
+    int i;
+    for(i = 0; (string_a[i] = string_b[i]) && (i < SOKUTEI_MAX_COUNTER_NAME_LENGTH); i++);
+    string_a[SOKUTEI_MAX_COUNTER_NAME_LENGTH] = '\0';
+    return string_a;
+}
+#define sokutei_string_equals(string_a, string_b) (sokutei_strcmp(string_a, string_b) == 0)
+
+///--- String functions
+
+
+/// Counters
+
+/**
+ * sokutei_counter_definitions is an array, which consists of the counter name followed by string terminator (\0), and the type of the counter.
+ */
+char sokutei_counter_definitions[SOKUTEI_MAX_COUNTER_COUNT][SOKUTEI_MAX_COUNTER_NAME_LENGTH + SOKUTEI_TYPE_INDICATOR_PADDING] = {'\0'};
+
+
+char sokutei_counters[SOKUTEI_MAX_COUNTER_COUNT * MAX_SIZE_OF_TYPES] = {0};
+int sokutei_number_of_counters = 0;
+
+///--- Counters
+
+
+/// Low level counter handling
+
+#define sokutei_counter_at_index(type, index) (type *)(sokutei_counters + (index * MAX_SIZE_OF_TYPES))
+
+
+char sokutei_get_type_of(const int index){
+    return sokutei_counter_definitions[index][SOKUTEI_TYPE_INDICATOR_INDEX];
+}
+
+
+int sokutei_get_index_of_counter(const char *counter_name){
+    int i;
+    for(i = 0; i < sokutei_number_of_counters; i++){
+        if(sokutei_string_equals(sokutei_counter_definitions[i], counter_name)){
+            return i;
+        }
+    }
+    return SOKUTEI_NOT_FOUND;
+}
+
+
+inline int sokutei_is_unknown_counter_type(const char type){
+    return !(type == SOKUTEI_INTEGER_TYPE || type == SOKUTEI_FLOAT_TYPE || type == SOKUTEI_INTERVAL_TYPE);
+}
+
+
+inline int sokutei_is_counter_limit_reached(){
+    return sokutei_number_of_counters + 1 >= SOKUTEI_MAX_COUNTER_COUNT;
+}
+
+
+inline int sokutei_create_new_counter(const char *counter_name, const char type){
+    sokutei_strcpy(sokutei_counter_definitions[sokutei_number_of_counters], counter_name);
+    sokutei_counter_definitions[sokutei_number_of_counters][SOKUTEI_TYPE_INDICATOR_INDEX] = type;
+    sokutei_number_of_counters++;
+    return 0;
+}
+
+
+int sokutei_add_counter(const char *counter_name, const char type){
+    if(sokutei_is_unknown_counter_type(type)){
+        return SOKUTEI_NOT_MATCHING_TYPE;
+    }
+
+    if(sokutei_is_counter_limit_reached()){
+        return SOKUTEI_COUNTER_LIMIT_REACHED;
+    }
+
+    if(sokutei_get_index_of_counter(counter_name) != SOKUTEI_NOT_FOUND){
+        return SOKUTEI_COUNTER_ALREADY_EXISTS;
+    }
+
+    return sokutei_create_new_counter(counter_name, type);
+}
+
+///--- Low level counter handling
+
+
+/// Iterations
+
+int sokutei_number_of_iterations =
+        SOKUTEI_MAX_MEASURED_ITERATIONS;
+
+int sokutei_current_iteration = 0;
+
+
+void sokutei_iteration_finish_handler(){
+    int index;
+    for(index = 0; index < (SOKUTEI_MAX_COUNTER_COUNT * MAX_SIZE_OF_TYPES); index++){
+        sokutei_counters[index] = 0;
+    }
```

Comment:
I think you can use [`memset`](https://en.cppreference.com/w/c/string/byte/memset) here.

Reply:
Thank you for the review @battila7! As far as I know, memset is a part of string.h header, which we decided not to include, for dependency reasons. Are you sure I should use it?

Reply:
Fuck this shit, you're right. I was a bit out-of-scope when reviewing 😄